qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -78,9 +74,15 @@ def __init__(
         if kak_basis_gate is not None:
             self.decomposer = TwoQubitBasisDecomposer(kak_basis_gate)
         elif basis_gates is not None:
-            self.decomposer = unitary_synthesis._decomposer_2q_from_basis_gates(
-                basis_gates, approximation_degree=approximation_degree
-            )
+            kak_gates = set(basis_gates or []).intersection(KAK_GATE_NAMES.keys())
+            kak_gate = None
+            if kak_gates:
+                kak_gate = KAK_GATE_NAMES[kak_gates.pop()]
+                self.decomposer = TwoQubitBasisDecomposer(
+                    kak_gate, basis_fidelity=approximation_degree or 1.0
+                )
+            if kak_gate is None:
+                self.decomposer = TwoQubitBasisDecomposer(CXGate())
```

Comment:
Can `kak_gate` actually be `None` if `kak_gates`  is truthy? If not, can we do this instead?

```suggestion
            if kak_gates:
                self.decomposer = TwoQubitBasisDecomposer(
                    KAK_GATE_NAMES[kak_gates.pop()], basis_fidelity=approximation_degree or 1.0
                )
            else:
                self.decomposer = TwoQubitBasisDecomposer(CXGate())
```